cache/src/main/scala/org/allenai/common/cache/JsonQueryCacheModule.scala

```diff
@@ -4,14 +4,25 @@ import org.allenai.common.guice.ConfigModule
 
 import com.google.inject.Provides
 import com.google.inject.name.Named
-import com.typesafe.config.{ Config, ConfigFactory }
+import com.typesafe.config.Config
 import redis.clients.jedis.{ JedisPool, JedisPoolConfig }
+import spray.json.JsonFormat
+
+/** A guice module for binding a `JsonQueryCache` instance with a parameterized value type. */
+class JsonQueryCacheModule[V](config: Config)(
+  implicit
+  jsonFormat: JsonFormat[V],
+  manifest: Manifest[V]
```

Comment:
The `Manifest` is what was needed to get the parameterized binding to work properly.  A little confusing since most of the existing Scala docs say `Manifest` is deprecated, but if you look in `Predef` you see it's been "un-deprecated".